core/src/filed/backup.cc

```diff
@@ -1602,11 +1613,15 @@ bool EncodeAndSendAttributes(JobControlRecord* jcr,
   Dmsg3(300, "File %s\nattribs=%s\nattribsEx=%s\n", ff_pkt->fname,
         attribs.c_str(), attribsEx);
 
-  {
-    std::unique_lock l(jcr->mutex_guard());
-    jcr->JobFiles++;                   /* increment number of files sent */
-    ff_pkt->FileIndex = jcr->JobFiles; /* return FileIndex */
+  if (std::optional findex = get_next_findex(jcr)) {
+    ff_pkt->FileIndex = *findex;
     PmStrcpy(jcr->fd_impl->last_fname, ff_pkt->fname);
+  } else {
+    Jmsg1(jcr, M_FATAL, 0,
```

Comment:
Currently this causes the backup to end in state `*** Backup Error ***`.  Should we maybe set the state to `JS_Incomplete` instead ?  Then the end state will show as `Backup failed -- incomplete`

Im not sure which one is better, what do you think @pstorz  ?